src/js/actions/layers.js

```diff
@@ -1377,45 +1382,147 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard 
+     * @param {Immutable.List.of<Layer>} artboards
+     * @return {Layer}
+     */
+
+    var _getRightMost = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+                if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                    return currentLayer;
+                } else {
+                    return selectedLayer;
+                }
+            }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     * @param {string} specs, {Immutable.List.of<Layer>} artboards, {Layer} layerRef
+     * @return {Bounds} finalBounds
+     */
+
+    var _getBoundsFromSpecs = function (specs, artboards, layerRef) {
+        var preset,
+            width, 
+            height,
+            rightmostlayer,
+            finalBounds;
+
+        //get the preset template from the specs
+        preset = specs.preset;
```

Comment:
It's better to declare and initialize in one step when it's not awkward to do so. This is a good candidate for combined declaration and initialization.